File: apple/Sources/ReactNativeWatchOS/Modifiers/PresentationDragIndicator.swift
import SwiftUI

/// SwiftUI `.presentationDragIndicator(_:)`. Sets the visibility of the
/// drag indicator on a sheet. `visibility` is parsed by
/// `RNWVisibilityParser` (automatic / visible / hidden).
///
/// This API is available on watchOS 9+, so it applies on every supported
/// deployment target — no `#available` fallback is needed.
enum RNWPresentationDragIndicatorModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("presentationDragIndicator") { view, params, _ in
            let visibility = RNWVisibilityParser.parse(params.string("visibility"))
            return AnyView(view.presentationDragIndicator(visibility))
        }
    }
}

File: src/modifiers/presentationDragIndicator.ts
import { createModifier } from './createModifier';

/// Visibility of a presentation's drag indicator. Mirrors SwiftUI's
/// `Visibility`.
export type DragIndicatorVisibility = 'automatic' | 'visible' | 'hidden';

/// SwiftUI `.presentationDragIndicator(_:)`. Sets the visibility of the
/// drag indicator on top of a sheet. Defaults to `'automatic'`, letting
/// the system decide. Available on watchOS 9+.
export function presentationDragIndicator(
  visibility: DragIndicatorVisibility = 'automatic'
) {
  return createModifier('presentationDragIndicator', { visibility });
}
